core/src/main/java/org/elasticsearch/rest/BytesRestResponse.java

```diff
@@ -136,8 +141,30 @@ private static XContentBuilder build(RestChannel channel, RestStatus status, Exc
 
         XContentBuilder builder = channel.newErrorBuilder().startObject();
         ElasticsearchException.generateFailureXContent(builder, params, e, channel.detailedErrorsEnabled());
-        builder.field("status", status.getStatus());
+        builder.field(STATUS, status.getStatus());
         builder.endObject();
         return builder;
     }
+
+    public static ElasticsearchStatusException errorFromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
+
+        ElasticsearchException exception = null;
+        RestStatus status = null;
+
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (STATUS.equals(parser.currentName()) == false) {
```

Comment:
shall we go back to saving the field_name to a variable now, so that we can check that the number we are parsing is effectively the status?